operators/test/e2e/failure_test.go

```diff
@@ -99,18 +106,77 @@ func TestKillOneMasterNode(t *testing.T) {
 }
 
 func TestKillSingleNodeReusePV(t *testing.T) {
-	// TODO :)
-	// This test cannot work until we correctly reuse PV between pods in the operator.
-	// We should not loose data, and ClusterUUID should stay the same
-
-	// s := stack.NewStackBuilder("test-failure-kill-single-node-no-pv").
-	// 	WithESMasterDataNodes(1, stack.DefaultResources).
-	//  WithPV().
-	// 	Stack
-	// matchNode := func(p corev1.Pod) bool {
-	// 	return true // match first node we find
-	// }
-	// killNodeTest(t, s, matchNode)
+	s := stack.NewStackBuilder("test-failure-pvc").
+		WithESMasterDataNodes(1, stack.DefaultResources)
+	matchNode := func(p corev1.Pod) bool {
+		return true // match first node we find
+	}
+	killNodeTest(t, s, helpers.ESPodListOptions(s.Elasticsearch.Name), matchNode)
+}
+
+func TestKillCorrectPVReuse(t *testing.T) {
+	s := stack.NewStackBuilder("test-failure-pvc").
```

Comment:
I'm wondering if we should maybe patch the storageClass **for this test only** (like: in the test).
Also skip this test if the k8s version we are testing against is <1.12.